### PR TITLE
some nss deadlock love

### PR DESCRIPTION
**Pull request by poettering**
Triggered by this thread:

https://lists.freedesktop.org/archives/systemd-devel/2018-July/040975.html

plus some other tiny microfixlets

### CODE REVIEW

File: doc/ENVIRONMENT.md [hunk 1]
```diff
@@ -96,3 +96,19 @@ systemd-timedated:
   NTP client services. If set, `timedatectl set-ntp on` enables and starts the
   first existing unit listed in the environment variable, and
   `timedatectl set-ntp off` disables and stops all listed units.
+
+systemd itself:
+
+* `$SYSTEMD_ACTIVATION_UNIT` — set for all NSS and PAM module invocatins that
```
Comment (lucab) [inline doc/ENVIRONMENT.md]: s/invocatins/invocations/
Also, do you want to codify here that the value is the name of the parent unit? (I'm not sure if it is already implicit now or intentionally left undefined)

File: src/core/execute.c [hunk 2]
```diff
@@ -2828,10 +2828,22 @@ static int exec_child(
                 }
         }
 
+        /* We are about to invoke NSS and PAM modules. Let's tell them what we are doing here, maybe they care. This is
+         * used by nss-resolve to disabled itself when we are about to start systemd-resolved, to avoid deadlocks. Note
```
Comment (keszybz) [inline src/core/execute.c]: disabled → disable

File: doc/ENVIRONMENT.md (hunk 1 above)
Comment (poettering) [reply doc/ENVIRONMENT.md]: hmm, what do you mean by "parent" unit? it's the unit that we run the NSS/PAM lookups for.

**Comment by poettering** [comment]
> But for stuff that we except to be always there, and running on a significant percentage of installations, removes the gains we get. Am I missing some important advantage apart from giving the DynamicUser implementation more testing?

Well it's sexier to use DynamicUser=1 ;-)

I figure this is a discussion to have, but it's kinda independent of this specific PR. Moreover, now that  that 5b5d82615011b9827466b7cd5756da35627a1608 we are basically back to static users for those three services anyway...

**Comment by poettering** [comment]
I force pushed a new version. I reworded the paragaph @lucab mentioned a bit, but I am not sure what @lucab precisely meant, hence I am not sure it improves anything for him... @lucab?

I also fixed the typo @keszybz found

File: doc/ENVIRONMENT.md (hunk 1 above)
Comment (lucab) [reply doc/ENVIRONMENT.md]: I was only asking if you want to specify in this doc what the value for the environment flag is supposed to be (as the modules may want  to check for specific values, like this PR does).

**Comment by poettering** [comment]
Ahh, I think I got @lucab's comment now. Will shortly push a new version that explicitly says the value is the unit's name. This was missing in the original version by mistake.

**Comment by poettering** [comment]
(just to clarify this: the new version i refer to in the comment is the one that is already here... for some reason github showed the message after the push, even though it happened the other way round)

**Pull request merged**